Solve 01-basics

Jenackler opened the pull request:
Solve 01-basics/10-create-app
Solve 01-basics/20-weather
Solve 02-basics-2/10-counter
Solve 02-basics-2/20-broken-map
Solve 02-basics-2/30-calculator
Solve 02-basics-2/40-marked-emails
Solve 02-basics-2/50-selected-meetup

Commit:
Solve 01-basics/10-create-app
Solve 01-basics/20-weather

jsru-1 commented:
Добавляю преподавателя (@ShGKme) для код-ревью.

Commit:
Merge remote-tracking branch 'refs/remotes/main/master'

Commit:
Merge remote-tracking branch 'refs/remotes/main/master'

Commit:
Solve 02-basics-2/10-counter
Solve 02-basics-2/20-broken-map
Solve 02-basics-2/30-calculator
Solve 02-basics-2/40-marked-emails
Solve 02-basics-2/50-selected-meetup

Jenackler committed:
Merge pull request #1 from js-tasks-ru/master

Solve 02-basics-2

jsru-1 commented:
Проверьте, пожалуйста, ваше решение, не все тесты прошли (PR не будет принят до тех пор, пока **все** добавленные задачи не будут решены).

Commit:
Merge remote-tracking branch 'origin/master'

jsru-1 commented:
Решение было обновлено, посмотрим что скажет @ShGKme

Jenackler committed:
Merge branch 'js-tasks-ru:master' into master

jsru-1 commented:
Проверьте, пожалуйста, ваше решение, не все тесты прошли (PR не будет принят до тех пор, пока **все** добавленные задачи не будут решены).

ShGKme commented on 01-basics/10-create-app/script.js:
```diff
@@ -1 +1,9 @@
+};
+createApp({
+  name: 'todayDate',
+  template: 'Сегодня ' + new Date().toLocaleDateString(navigator.language, options),
```
Так определять шаблон некорректно. Здесь не инструментами Vue шаблоне описано, что должно выводиться, а сгенерирован сам шаблон на JS. Вместо этого шаблон должен быть фиксированной строкой, и уже внутри шаблона можно использовать директивы и текстовую интерполяцию для определения содержимого.

- Так собрать шаблон в основном формате Vue компонентов - SFC - вообще не возможно
- Значение шаблона определяется в момент описания компонента, а не в момент его работы. Так, если страница будет открыта сегодня, но компонент впервые срендерится только завтра - он всё равно выведет вчерашнюю дату

ShGKme replied:
Если это было сделано, потому что `navigator.language` не доступен в шаблоне - лучше использовать переменную/функцию, описанную в `setup`.

Jenackler replied:
Загрузил исправления

Commit:
Solve 03-components/10-MeetupView
Solve 03-components/20-UiClock
Solve 03-components/30-removable-emails
Solve 03-components/40-UiCounter
Solve 03-components/50-weather-components
Fix 01-basics/10-create-app
Fix 01-basics/20-weather

jsru-1 commented:
Добавляю преподавателя (@ShGKme) для код-ревью.

Jenackler commented:
Исправлены указанные ошибки.
Добавлено решение для задач по компонентам

Commit:
Solve 03-components/10-MeetupView
Solve 03-components/20-UiClock
Solve 03-components/30-removable-emails
Solve 03-components/40-UiCounter
Solve 03-components/50-weather-components
Fix 01-basics/10-create-app
Fix 01-basics/20-weather

jsru-1 commented:
Решение было обновлено, посмотрим что скажет @ShGKme

Commit:
Solve 03-components/10-MeetupView
Solve 03-components/20-UiClock
Solve 03-components/30-removable-emails
Solve 03-components/40-UiCounter
Solve 03-components/50-weather-components
Fix 01-basics/10-create-app
Fix 01-basics/20-weather

jsru-1 commented:
Решение было обновлено, посмотрим что скажет @ShGKme

Jenackler committed:
Merge branch 'js-tasks-ru:master' into master

jsru-1 commented:
Проверьте, пожалуйста, ваше решение, не все тесты прошли (PR не будет принят до тех пор, пока **все** добавленные задачи не будут решены).

Commit:
Solve 04-sfc/10-MeetupView-script-setup
Solve 04-sfc/20-MeetupCover-style-v-bind
Solve 04-sfc/30-UiStretch

jsru-1 commented:
Решение было обновлено, посмотрим что скажет @ShGKme

ShGKme reviewed (approved):
Все задачи приняты

ShGKme commented on 03-components/50-weather-components/components/weatherCard/WeatherCard.js:
```diff
@@ -0,0 +1,45 @@
+import WeatherAlert from './weatherAlert/WeatherAlert.js'
+import WeatherCardInfo from './weatherCardInfo/WeatherCardInfo.js'
+
+export default defineComponent({
+  name: 'WeatherCard',
+
+  props: {
+    card : {
+      type: Object,
+      required: true
+    },
+    icons: {
+      type: Object,
+      required: true
+    }
+  },
```
Пропустил при проверке.

Сейчас для использования многих компонентов (включая компонент списка карточек) нужно передавать массив иконок. Это неудобно - пользователю даже списка, нужно иметь этот список, его кастомизации пока не требуется. Такая возможность может быть плюсом, если она дополнительная и опциональная (есть иконки по умолчанию). Но это неудобный интерфейс компонента, если для вывода списка нужно передавать объект иконок для разных погодных условий.

ShGKme commented on 03-components/50-weather-components/WeatherApp.js:
```diff
@@ -1,52 +1,25 @@
-          </div>
-        </li>
-      </ul>
+      <WeatherCardList :cardsList="wData" :icons="icons"/>
```
По стайл гайду принято пропсы (и события) в шаблоне использовать в `kebab-case`
